solidity/contracts/vortex/VortexBurner.sol

```diff
@@ -0,0 +1,522 @@
+        uint256[] memory amounts = new uint256[](tokens.length);
+
+        for (uint256 i = 0; i < tokens.length; ++i) {
+            IERC20 token = tokens[i];
+
+            address[] memory path = new address[](3);
+            path[0] = address(token);
+
+            // don't look up for a converter for either the network or the governance token, since they are going to be
+            // handled in a special way during the burn itself
+            if (token != _networkToken && token != _govToken) {
+                path[1] = address(networkTokenConverterAnchor(token, registry));
+                path[2] = address(_networkToken);
+            }
+
+            paths[i] = path;
+
+            // make sure to retrieve the balance of either an ERC20 or an ETH reserve
+            if (token == NATIVE_TOKEN_ADDRESS) {
+                amounts[i] = address(feeWallet).balance;
+            } else {
+                amounts[i] = token.balanceOf(address(feeWallet));
+            }
+        }
+
+        // get the governance token converter path
+        address[] memory govPath = new address[](3);
+        govPath[0] = address(_networkToken);
+        govPath[1] = address(networkTokenConverterAnchor(_govToken, registry));
+        govPath[2] = address(_govToken);
+
+        return (paths, amounts, govPath);
+    }
+
+    /**
+     * @dev applies the burn incentive fee on the provided amount and returns the net amount and the fee
+     *
+     * @param amount the network tokens amount
+     *
+     * @return the net amount
+     * @return the fee amount
+     */
+    function applyIncentiveFee(uint256 amount) private view returns (uint256, uint256) {
+        uint256 fee = _burnIncentiveFee;
+        if (fee == 0) {
+            return (amount, 0);
+        }
+
+        uint256 incentiveAmount = Math.min(amount.mul(fee).div(PPM_RESOLUTION), _maxBurnIncentiveFeeAmount);
+
+        return (amount.sub(incentiveAmount), incentiveAmount);
+    }
+
+    /**
+     * @dev finds the converter anchor of the 50/50 standard pool converter between the specified token and the network token
+     *
+     * @param token the source token
+     * @param converterRegistry the converter registry
+     *
+     * @return the converter anchor of the 50/50 standard pool converter between the specified token
+     */
+    function networkTokenConverterAnchor(IERC20 token, IConverterRegistry converterRegistry)
+        private
+        view
+        returns (IConverterAnchor)
+    {
+        // initialize both the source and the target tokens
+        IERC20[] memory reserveTokens = new IERC20[](2);
+        reserveTokens[0] = _networkToken;
+        reserveTokens[1] = token;
+
+        // make sure to only look up for 50/50 converters
+        uint32[] memory standardReserveWeights = new uint32[](2);
+        standardReserveWeights[0] = STANDARD_POOL_RESERVE_WEIGHT;
+        standardReserveWeights[1] = STANDARD_POOL_RESERVE_WEIGHT;
+
```

Comment:
Minor: I'd use `PPM_RESOLUTION / 2` here instead of declaring another constant.

Reply:
That would be a code duplication we want to generally avoid